### PR TITLE
Use local SSDs and format using startup script.

### PR DESCRIPTION
**Pull request by mberhault**
Other tweaks:
* reduce boot disk size (50G -> 10G)
* fix name parsing

### CODE REVIEW

File: cluster.go [hunk 1]
```diff
@@ -70,10 +70,10 @@ func (vl VMList) Less(i, j int) bool { return vl[i].Name < vl[j].Name }
 
 func namesFromVMName(name string) (string, string, error) {
 	parts := strings.Split(name, "-")
-	if len(parts) != 3 {
+	if len(parts) < 3 {
 		return "", "", fmt.Errorf("expected VM name in the form %s, got %s", vmNameFormat, name)
 	}
-	return parts[0], parts[0] + "-" + parts[1], nil
+	return parts[0], strings.Join(parts[0:len(parts)-1], "-"), nil
```
Comment (petermattis) [inline cluster.go]: FYI, the `0:` isn't necessary. This could be `strings.Join(parts[:len(parts)-1], "-")`.

File: utils.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+func writeStartupScript() (string, error) {
+	tmpfile, err := ioutil.TempFile("", "gce-startup-script")
+	if err != nil {
+		return "", err
+	}
+
+	if _, err := tmpfile.WriteString(gceLocalSSDStartupScript); err != nil {
+		return "", err
+	}
+	if err := tmpfile.Close(); err != nil {
```
Comment (petermattis) [inline utils.go]: The common pattern is to add a `defer tmpfile.Close()` immediately after opening the file.

File: utils.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"io/ioutil"
+)
+
+// Startup script used to find/format/mount all local SSDs in GCE.
+// Each disk is mounted to /mnt/data<disknum> and chmoded to all users.
+const gceLocalSSDStartupScript = `
+#!/usr/bin/env bash
```
Comment (petermattis) [inline utils.go]: I thought this had to be the first line of the file. This is making it the second.

File: gcloud.go [hunk 4]
```diff
@@ -91,10 +99,13 @@ func createVMs(names []string, lifetime time.Duration) error {
 		"--scopes", "https://www.googleapis.com/auth/devstorage.read_only,https://www.googleapis.com/auth/logging.write,https://www.googleapis.com/auth/monitoring.write,https://www.googleapis.com/auth/servicecontrol,https://www.googleapis.com/auth/service.management.readonly,https://www.googleapis.com/auth/trace.append",
 		"--image", "ubuntu-1604-xenial-v20171002",
 		"--image-project", "ubuntu-os-cloud",
-		"--boot-disk-size", "50",
+		"--boot-disk-size", "10",
 		"--boot-disk-type", "pd-ssd",
+		"--local-ssd", "interface=SCSI",
```
Comment (petermattis) [inline gcloud.go]: Did you want to make this optional?

File: cluster.go (hunk 1 above)
Comment (mberhault) [reply cluster.go]: Done

File: gcloud.go (hunk 4 above)
Comment (mberhault) [reply gcloud.go]: Yup. Done, as well as `--machine-type`. Added a `VMOpts` to pass everything around.

File: utils.go (hunk 3 above)
Comment (mberhault) [reply utils.go]: Still works, but yes, you're right. Done.

File: utils.go (hunk 2 above)
Comment (mberhault) [reply utils.go]: Done.

**Pull request merged**